agg/Color.cs

```diff
@@ -1233,13 +1233,9 @@ public static double Contrast(this IColorType thisA, IColorType color)
 
 		public static ColorF WithLightness(this IColorType original, double lightness)
 		{
-			double hue0To1;
-			double saturation0To1;
-			double lightness0To1;
+			ColorF colorF = original.ToColorF();
```

Comment:
ToColorF is optimized in ColorF to return this, no conditional and/or cast needed